tests/testsuite/inheritable_workspace_fields.rs

```diff
@@ -1234,11 +1234,13 @@ fn error_workspace_false() {
 [ERROR] failed to parse manifest at `[CWD]/Cargo.toml`
 
 Caused by:
-  TOML parse error at line 7, column 41
+  TOML parse error at line 7, column 27
     |
   7 |             description = { workspace = false }
-    |                                         ^^^^^
+    |                           ^^^^^^^^^^^^^^^^^^^^^
   `workspace` cannot be false
+  in `workspace`
+
```

Comment:
@dtolnay In this commit, I switched some explicit visitors to `serde_untagged` and it seems the error handling isn't 1-1.  This error message is basically showing that `toml` deserialized `workspace = false` and generated an error that then got turned around and plugged into `Error::custom`, causing us to wrap that `toml` error in another `toml` error but with context this time. 
- The context is for a wider span
- We then show the redundant `in workspace` which came from the original error
- We get two newlines, one for each `toml` Error

Reply:
I am working on a fix.

Reply:
I've split this change out of the PR